src/components/Toolbar/index.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Menu, { Item } from 'components/Menu'
+import styles from './index.styl'
+
+import { withBreakpoints } from 'cozy-ui/react'
+
+const { BarRight } = cozy.bar
+
+class Toolbar extends Component {
+  render() {
+    const {
+      children,
+      moreMenuTitle,
+      moreMenuDisabled,
+      moreMenuClassName,
+      moreMenuButton,
+      breakpoints: { isMobile }
+    } = this.props
+
+    const visibleActions = React.Children.toArray(children).filter(
+      child => child.props.visible
+    )
+    const highlightedActions = visibleActions.filter(
+      child => child.props.highlighted
+    )
+    const secondaryActions = visibleActions.filter(
+      child => !child.props.highlighted
+    )
+
+    const MoreMenu = (
+      <Menu
+        title={moreMenuTitle}
+        disabled={moreMenuDisabled}
+        className={moreMenuClassName}
+        button={moreMenuButton}
+      >
+        {highlightedActions.map(component =>
+          this.wrapInItem(this.renderToolbarAction(component, false, false))
+        )}
+        {secondaryActions.map(component =>
+          this.wrapInItem(this.renderToolbarAction(component, false, true))
+        )}
+      </Menu>
+    )
+
+    return (
+      <div role="toolbar">
+        {highlightedActions.map(component =>
+          this.renderToolbarAction(component, true, false)
+        )}
+        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
+      </div>
+    )
+  }
+
+  renderToolbarAction(ToolbarAction, forcedHighlightProp, isSecondaryAction) {
+    try {
+      const children = ToolbarAction.props.children
+      const child = React.cloneElement(React.Children.only(children), {
+        highlighted: forcedHighlightProp
+      })
+
+      if (isSecondaryAction) return child
+      else {
+        const hiddenStyle = forcedHighlightProp
+          ? styles['u-hide--mob']
+          : styles['u-hide--desk']
+        return <span className={hiddenStyle}>{child}</span>
+      }
+    } catch (_) {
+      return null
+    }
+  }
+
+  wrapInItem(component) {
+    return <Item>{component}</Item>
+  }
+}
+
+Toolbar.propTypes = {
+  moreMenuTitle: PropTypes.string.isRequired,
+  moreMenuDisabled: PropTypes.bool,
+  moreMenuClassName: PropTypes.string,
+  moreMenuButton: PropTypes.element.isRequired
+}
+
+Toolbar.defaultProps = {
+  moreMenuDisabled: false,
+  moreMenuClassName: ''
+}
+
+export class ToolbarAction extends Component {} // needs to extend Component to have `props.children`
```

Comment:
This is what bothers me the most with the current implementation — `ToolbarAction ` is just a lie.
We use it as a way to declare all the actions and [be able to filter them](https://github.com/cozy/cozy-drive/pull/679/files#diff-cd8ecf10d1445222404e50d7f56d783fR21). But they are [never even rendered](https://github.com/cozy/cozy-drive/pull/679/files#diff-cd8ecf10d1445222404e50d7f56d783fR59) because we are only interested in the children.

@enguerran suggested using a HOC, since really, it's a HOC. But eventually i didn't do it because it changes the API from:

```
<Toolbar>
  <ToolbarAction visible highlighted={false}>
    <MyComponent />
  </ToolbarAction>
</Toolbar>
```

which is nice and hierchical, to something like

```
<Toolbar>
  {asToolbarAction(<MyComponent />, true, false)}
</Toolbar>
```